src/Effect.cpp

```diff
@@ -298,28 +298,25 @@ Effect::add_external_effect(const Effect &e, std::vector<const Block*> call_chai
 }
 
 /*
- * TODO: tried doing this with `find', but I couldn't get all the `const's
- * right.  Gave up; wrote it by hand!
+ * 
+ *
  */
 bool
 Effect::is_read(const Variable *v) const
 {
-	vector<Variable *>::size_type len = read_vars.size();
-	vector<Variable *>::size_type i;
-
-	for (i = 0; i < len; ++i) {
-		if (read_vars[i] == v) {
-			return true;
+	if (std::find(read_vars.begin(), read_vars.end(), v) != read_vars.end()) {
+		return true;
+	}
+	else {
```

Comment:
Seems we don't need the ```else``` branch, and hence we don't need to change the code below?